cfme/scripting/setup_env.py

```diff
@@ -0,0 +1,113 @@
+import sys
+import click
+from cfme.test_framework.sprout.client import SproutClient
+from collections import namedtuple
+from utils.conf import credentials, cfme_data
+from wait_for import wait_for
+
+
+def provision_appliances(count, cfme_version, provider, lease_time):
+    sprout_client = SproutClient.from_config()
+    apps, request_id = sprout_client.provision_appliances(version=str(cfme_version),
+        count=count, preconfigured=False, lease_time=lease_time, provider=provider)
+    return apps
+
+
+@click.command('setup-env', help='Provisions and configures appliances')
+@click.option('--cfme-version', required=True)
+@click.option('--lease-time', default=180)
+@click.option('--ha', 'mode', flag_value='ha')
+@click.option('--distributed', 'mode', flag_value='distributed')
+@click.option('--replication', 'mode', flag_value='replication')
```

Comment:
`ha`, `distributed` and `replication` shouldn't be options, they should be a group commands. See `cfme/scripting/appliance.py`